File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.go
func maxProfit(prices []int) int {
    left := 0
    maxProfit := 0
    for right := 1; right < len(prices); right++{
        if prices[left] < prices[right]{
            profit := prices[right] - prices[left]
            if profit > maxProfit{
                maxProfit = profit
            }
        }else {
            left = right
        }
    }
    return maxProfit
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.java
class Solution {
     public  boolean validPalindrome(String s) {
        int left = 0;
        int right = s.length() -1;
        int counter = 0;
        while (left <= right){
            if (s.charAt(left) == s.charAt(right)){
                left++;
                right--;
            }else{
                return palindrome(left+1, right, s)|| palindrome(left,right-1,s);
            }
        }
        return true;
    }
    public boolean palindrome(int l, int r, String str){
        while (l <= r){
            if (str.charAt(l) != str.charAt(r)){
                return false;
            };
            l++;
            r--;
        }
        return true;
    }
}